pkg/jobs/jobspb/jobs.proto

```diff
@@ -171,6 +171,38 @@ message DroppedTableDetails {
   Status status = 3;
 }
 
+// WaitingForGCDetails is the payload of the job which waits for the TTL of
+// schema elements to expire and then GCs them.
+// These should store the drop times.
+message WaitingForGCDetails {
+  message DroppedIndex {
+    int64 index_id = 1 [(gogoproto.customname) = "IndexID",
+                       (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/sqlbase.IndexID"];
+    int64 drop_time = 2;
+    int64 deadline = 3;
+  }
+
+  message DroppedID {
+    int64 id = 1 [(gogoproto.customname) = "ID",
+                 (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/sqlbase.ID"];
+    // DropTime is used re-calculate the deadline when the zone config was updated.
+    int64 drop_time = 2;
```

Comment:
Context: The old schema changer maintained a deadline for everything it was going to GC. We then decide if something is ready to GC by just looking at it's deadline. I've keep the same thing here.

The idea that I went with here is that the job details should know all the information it needs to GC, and should be the source of truth for this information (rather than the table descriptor). This leads to a bit of wonkiness with the table's drop time being persisted on the table descriptor and the job's details.

So in summary the options are:
1. Keep the drop time on the table descriptor and then update the deadline in the jobs details to keep track when it is time to GC.
2. Only keep the IDs of the indexes/tables to drop in the job's details, and read the info off the table descriptor every time.
3. The current implementation, which expects all the relevant information from the table descriptor to be loaded at creation time and doesn't need to interact with the table descriptor further.

Question: Do you strongly prefer one of the three options listed?